crates/subspace-farmer/src/bin/subspace-farmer/commands/farm.rs

```diff
@@ -119,98 +120,80 @@ pub(crate) async fn farm_multi_disk(
-            enable_dsn_archiving: matches!(archiving, ArchivingFrom::Dsn),
-            enable_dsn_sync: dsn_sync,
-            enable_farming: !disable_farming,
-            plot_factory: move |options: PlotFactoryOptions<'_>| {
-                Plot::open_or_create(
-                    options.single_plot_farm_id,
-                    options.plot_directory,
-                    options.metadata_directory,
-                    options.public_key,
-                    options.max_plot_size,
-                )
-            },
-        })
-        .await?;
-
-        single_disk_farms.push(single_disk_farm);
+        })?;
+
+        single_disk_plots.push(single_disk_plot);
     }
 
     let record_size = record_size.expect("At least one farm is always present, checked above; qed");
     let recorded_history_segment_size = recorded_history_segment_size
         .expect("At least one farm is always present, checked above; qed");
 
-    // Start RPC server
-    let ws_server = match WsServerBuilder::default()
-        .build(ws_server_listen_addr)
-        .await
-    {
-        Ok(ws_server) => ws_server,
-        Err(jsonrpsee::core::Error::Transport(error)) => {
-            warn!(
-                address = %ws_server_listen_addr,
-                %error,
-                "Failed to start WebSocket RPC server on, trying random port"
-            );
-            ws_server_listen_addr.set_port(0);
-            WsServerBuilder::default()
-                .build(ws_server_listen_addr)
-                .await?
-        }
-        Err(error) => {
-            return Err(error.into());
-        }
-    };
-    let ws_server_addr = ws_server.local_addr()?;
-    let rpc_server = RpcServerImpl::new(
-        record_size.get(),
-        recorded_history_segment_size,
-        Arc::new(
-            single_disk_farms
-                .iter()
-                .map(|single_disk_farm| single_disk_farm.piece_getter())
-                .collect::<Vec<_>>(),
-        ),
-        Arc::new(
-            single_disk_farms
-                .iter()
-                .flat_map(|single_disk_farm| {
-                    single_disk_farm
-                        .single_plot_farms()
-                        .iter()
-                        .map(|single_plot_farm| single_plot_farm.object_mappings().clone())
-                })
-                .collect::<Vec<_>>(),
-        ),
-    );
-    let _ws_server_guard = CallOnDrop::new({
-        let ws_server = ws_server.start(rpc_server.into_rpc())?;
-        let tokio_handle = tokio::runtime::Handle::current();
-
-        move || {
-            if let Ok(waiter) = ws_server.stop() {
-                tokio::task::block_in_place(move || tokio_handle.block_on(waiter));
-            }
-        }
-    });
-
-    info!("WS RPC server listening on {ws_server_addr}");
-
-    let mut single_disk_farms_stream = single_disk_farms
+    // TODO: Restore functionality or remove
+    // // Start RPC server
```

Comment:
I'm already working on a separate PR where I do the cleanup. I think I'll just remove RPC server there completely and move the reconstruction logic into standalone reusable piece of code instead. If we need something like this in the future, it should be a separate application (some sort of gateway). It only worked initially when the farmer stored all the data in existence, as soon as blockchain history doesn't fit anymore, it stops working properly because only local data is used.